testhelper/integrationtesthelper.go

```diff
@@ -85,62 +179,185 @@ func IntegrationTimedLidar(
-		defer timeTracker.Mu.Unlock()
+		defer timeTracker.mu.Unlock()
 		/*
 			Holds the process until for all necessary IMU data has been sent to cartographer. Only applicable
 			when the IMU is present (timeTracker.NextImuTime has been defined) and is always true in the first iteration.
 			This and the manual definition of timestamps allow for consistent results.
 		*/
 		for {
-			timeTracker.Mu.Lock()
-			if timeTracker.ImuTime == defaultTime {
+			timeTracker.mu.Lock()
+			if timeTracker.imuTime == defaultTime {
 				time.Sleep(sensorDataIngestionWaitTime)
 				break
 			}
 
-			if i <= 1 || timeTracker.LidarTime.Sub(timeTracker.ImuTime) <= 0 {
+			if i <= 1 || timeTracker.lidarTime.Sub(timeTracker.imuTime) <= 0 {
 				time.Sleep(sensorDataIngestionWaitTime)
 				break
 			}
-			timeTracker.Mu.Unlock()
+			timeTracker.mu.Unlock()
 		}
 
 		// Communicate that all lidar readings have been sent to cartographer or if the last IMU reading has been sent,
 		// checks if LastLidarTime has been defined. If so, simulate endOfDataSet error.
-		t.Logf("TimedLidarReading Mock i: %d, closed: %v, readingTime: %s\n", i, closed, timeTracker.LidarTime.String())
-		if i >= NumPointClouds || timeTracker.LastImuTime != defaultTime {
+		t.Logf("TimedLidarReading Mock i: %d, closed: %v, readingTime: %s\n", i, closed, timeTracker.lidarTime.String())
+		if i >= NumPointClouds || timeTracker.lastImuTime != defaultTime {
 			// Sends a signal to the integration sensor's done channel the first time end of dataset has been sent
 			if !closed {
 				done <- struct{}{}
 				closed = true
-				timeTracker.LastLidarTime = timeTracker.LidarTime
+				timeTracker.lastLidarTime = timeTracker.lidarTime
 			}
 
 			return s.TimedLidarReadingResponse{}, replaylidar.ErrEndOfDataset
 		}
 
 		// Get next lidar data
-		resp, err := createTimedLidarReadingResponse(t, i, replay, timeTracker)
+		resp, err := createTimedLidarReadingResponse(t, i, timeTracker)
 		if err != nil {
 			return resp, err
 		}
 
 		// Advance the data index and update time tracker (manual timestamps occurs here)
 		i++
-		timeTracker.LidarTime = timeTracker.LidarTime.Add(sensorReadingInterval)
-		timeTracker.NextLidarTime = timeTracker.LidarTime.Add(sensorReadingInterval)
+		timeTracker.lidarTime = timeTracker.lidarTime.Add(sensorReadingInterval)
+		timeTracker.nextLidarTime = timeTracker.lidarTime.Add(sensorReadingInterval)
 
 		return resp, nil
 	}
 
 	return injectLidar, nil
 }
 
-// IntegrationTimedIMU returns a mock timed IMU sensor.
+// IntegrationCartographer is responsible for running a viam-cartographer process using the desired mock sensors. Once started it will
+// wait for all data to be processed by monitor sensor channels. After data has been fully processed, the endpoints Position,
+// PointCloudMap, InternalState are evaluated and the process is closed out. The final internal state of cartographer is then returned.
+func IntegrationCartographer(
```

Comment:
Moved this from `integration_test.go` here and renamed it from `testHelperCartographer` to `IntegrationCartographer`.